libvast/vast/format/json.hpp

```diff
@@ -104,7 +104,7 @@ struct default_selector {
     return "json-reader";
   }
 
-  detail::flat_map<std::vector<std::string>, record_type> type_cache = {};
+  detail::stable_map<std::vector<std::string>, record_type> type_cache = {};
```

Comment:
I have no opinion on the change itself, but this seems to be a bit more than "fixing a warning".

Can we pull this into a separate commit that includes the reasoning in the commit message (and maybe also before/after benchmarking results if performance was the reason for changing the data structure)

Reply:
That was supposed to be one commit further up. Will rebase in a sec. It's actually a necessity, it was slightly broken.

Reply:
Sounds good then, the idea has to be "make it correct" and only then "make it fast".

Let's still take a close look at the next set of benchmark results, to see if this is introducing a performance regression or if it doesnt matter.